### PR TITLE
Separate preview from deploy

### PR DESCRIPTION
**Pull request by amitizle**

### CODE REVIEW

File: packages/cli/src/commands/deploy.ts [hunk 1]
```diff
@@ -139,14 +141,16 @@ export class DeployCommand implements CliCommand {
...
-    const result = await deploy(
-      workspace,
-      shouldDeploy(this.stdout, workspace),
-      (item: PlanItem, step: ItemStatus, details?: string) =>
-        this.updateAction(item, step, details),
-      this.inputServices,
-      this.force
-    )
+    const actionPlan = await preview(workspace, this.inputServices)
+    const result = (this.force || await shouldDeploy(this.stdout, workspace)(actionPlan))
```
Comment (hadard) [inline packages/cli/src/commands/deploy.ts:145]: IMO `shouldDeploy` should be moved from `callbacks.ts` to this file (not a callback anymore)

Comment (ori-moisis) [reply packages/cli/src/commands/deploy.ts:145]: it also doesn't make sense IMO that `shouldDeploy` gets two parameters a returns a function that gets the third parameter when it is used like this, why not just get all three parameters?

on the more nit-picky side I don't like that the printing of the plan is a side effect of `shouldDeploy`, by reading this code I wouldn't know the plan is ever shown to the user which is strange when it is in the context of the cli where the code should mostly tell the story of the user interaction (I understand why it had to be like that when this was a callback but now that it isn't a callback I think we can do better)

Comment (amitizle) [reply packages/cli/src/commands/deploy.ts:145]: @ori-moisis totally agree on the plan printing

File: packages/cli/test/commands/deploy.test.ts [hunk 2]
```diff
@@ -123,6 +130,38 @@ describe('deploy command', () => {
     })
   })
 
+  describe('should deploy user input', async () => {
+    let content: string
+    // let mockGetUserBooleanInput: jest.Mock
```
Comment (ori-moisis) [inline packages/cli/test/commands/deploy.test.ts:135]: nit: remove commented out code

Comment (amitizle) [reply packages/cli/test/commands/deploy.test.ts:135]: not nit at all :) will remove, thanks!

**Pull request merged**